### PR TITLE
Update triton xpu commit pin

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137886
* #135567

# Motivation
Due to the code change of https://github.com/pytorch/pytorch/pull/135567, triton-xpu needs to fetch `tensor.data_ptr()` via `uint64` instead of `int64`, refer to https://github.com/intel/intel-xpu-backend-for-triton/pull/2192

# Additional Context
triton commit comes from release branch: https://github.com/intel/intel-xpu-backend-for-triton/tree/release/3.2.0

### CODE REVIEW

File: .github/workflows/build-triton-wheel.yml [hunk 1]
```diff
@@ -118,6 +118,7 @@ jobs:
           docker exec -t "${container_name}" "${PYTHON_EXECUTABLE}"  -m pip install -U setuptools==67.4.0
           # Triton xpu build use GCC11
           if [[ "${BUILD_DEVICE}" == xpu ]]; then
+            docker exec -t "${container_name}" "${PYTHON_EXECUTABLE}"  -m pip install -U pybind>=2.13.1
```
Comment (guangyey) [inline .github/workflows/build-triton-wheel.yml:121]: triton xpu depends on pybind.

Comment (anmyachev) [inline .github/workflows/build-triton-wheel.yml:121]: @guangyey ?
```suggestion
            docker exec -t "${container_name}" "${PYTHON_EXECUTABLE}"  -m pip install -U pybind11>=2.13.1
```

https://github.com/intel/intel-xpu-backend-for-triton/blob/8d5374a3114fb922c4d8c7e2bb646ef24c1dee2b/python/pyproject.toml#L3C77-L3C93

Comment (guangyey) [reply .github/workflows/build-triton-wheel.yml]: Thanks.

**Comment by guangyey** [comment]
These failures are related with https://github.com/intel/intel-xpu-backend-for-triton/issues/2488.

File: .github/scripts/build_triton_wheel.py [hunk 2]
```diff
@@ -162,7 +162,8 @@ def build_triton(
 
         # change built wheel name and version
         env["TRITON_WHEEL_NAME"] = triton_pkg_name
-        env["TRITON_WHEEL_VERSION_SUFFIX"] = version_suffix
+        if device != "xpu":
+            env["TRITON_WHEEL_VERSION_SUFFIX"] = version_suffix
```
Comment (guangyey) [inline .github/scripts/build_triton_wheel.py:166]: Avoid to duplicated git commit suffix, refer to https://github.com/triton-lang/triton/pull/4812/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfR650

**Comment by guangyey** [comment]
These failures are related to https://github.com/pytorch/pytorch/issues/141466, and we have filed a PR https://github.com/pytorch/pytorch/pull/141470 to fix it.

**Comment by guangyey** [comment]
@EikanWang May I know if it is reasonable to update xpu triton commit pin.

**Comment by EikanWang** [comment]
@guangyey , yes, it is.

**Review by EikanWang — APPROVED** [review]

**Comment by guangyey** [comment]
Following https://github.com/pytorch/pytorch/pull/139206, we will land this PR to update triton xpu to 3.2

**Review by atalman — APPROVED** [review]
lgtm.

**Pull request closed without merging**